eth/executionclient/execution_client_test.go

```diff
@@ -189,29 +184,40 @@ func TestStreamLogs(t *testing.T) {
 	}
 
 	// Wait for blocksWithLogsLength-followDistance blocks to be streamed.
-Unfollowed:
+Wait1:
 	for {
 		select {
 		case <-ctx.Done():
-			require.Equal(t, int64(blocksWithLogsLength-followDistance), streamedLogsCount.Load())
+			require.Failf(t, "timed out", "err: %v, streamedLogsCount: %d", ctx.Err(), streamedLogsCount.Load())
 		case <-time.After(time.Millisecond * 5):
 			if streamedLogsCount.Load() == int64(blocksWithLogsLength-followDistance) {
-				break Unfollowed
+				break Wait1
 			}
 		}
 	}
```

Comment:
So, (in `stage` branch) if we drop into `case <-ctx.Done()` here this `for` loop will very likely spin perpetually (allocating new `Timer` on every iteration through `time.After`).

Reply:
great catch